EngineLayer/FdrAnalysis/FdrAnalysisEngine.cs

```diff
@@ -207,55 +156,34 @@ private void DoFalseDiscoveryRateAnalysis(FdrAnalysisResults myAnalysisResults)
             if (AnalysisType == "PSM")
             {
                 CountPsm();
-            }
-        }
-
-        private static double GetEValue(PeptideSpectralMatch psm, int globalMeanCount, double globalMeanScore, out double maximumLikelihood)
-        {
-            // get all of the PSM's scores for all hits, sort them, then remove the last value (the best score)
-            List<double> scoresWithoutBestHit = new List<double>();
-            scoresWithoutBestHit.AddRange(psm.AllScores);
-            scoresWithoutBestHit.Sort();
-
-            if (scoresWithoutBestHit.Any())
-            {
-                scoresWithoutBestHit.RemoveAt(scoresWithoutBestHit.Count - 1);
-            }
-
-            // this is the "default" case for when there are no scores except the best hit
-            // it uses a global mean score (all scores across all PSMs) to generate the null Poisson distribution
-            // this will be overriden by the next few lines if there are enough scores in this PSM to estimate a null distribution
-            double preValue = SpecialFunctions.GammaLowerRegularized(globalMeanScore, psm.Score);
-            maximumLikelihood = globalMeanScore;
-
-            // calculate single-spectrum evalue if there are enough hits besides the best scoring peptide
-            if (psm.Score == 0)
-            {
-                preValue = 1;
-                maximumLikelihood = 0;
-            }
-            else if (scoresWithoutBestHit.Any())
-            {
-                maximumLikelihood = scoresWithoutBestHit.Average();
-
-                // this is the cumulative distribution for the poisson at each score up to but not including the score of the winner.
-                // This is the probability that the winner has of getting that score at random by matching against a SINGLE spectrum
-                if (maximumLikelihood > 0)
+                if (AllPsms.Count > 0)
                 {
-                    preValue = SpecialFunctions.GammaLowerRegularized(maximumLikelihood, psm.Score);
+                    myAnalysisResults.BinarySearchTreeMetrics = PEP_Analysis.ComputePEPValuesForAllPSMsGeneric(AllPsms);
+                    Compute_PEPValue_Based_QValue(AllPsms);
                 }
             }
 
-            // Now the probability of getting the winner's score goes up for each spectrum match.
-            // We multiply the preValue by the number of theoretical spectrum within the tolerance to get this new probability.
-            int count = scoresWithoutBestHit.Count;
-            if (count == 0)
+            if (AnalysisType == "Peptide")
             {
-                count = globalMeanCount;
+                Compute_PEPValue_Based_QValue(AllPsms);
             }
+        }
 
-            double probabilityOfScore = 1 - Math.Pow(preValue, count);
-            return count * probabilityOfScore;
+        public static void Compute_PEPValue_Based_QValue(List<PeptideSpectralMatch> psms)
```

Comment:
Is my understanding right? Here PEPValue is just used as a score range from 0-1 instead of probability. Could we use PEP < 0.01 instead of FDR < 0.01?

Reply:
no. it is an actual probability. it is the probability that an individual PSM is not not a true positive. We have not had anything like it in metamorpheus so far. FDR is a groupwise number. 1%FDR means that 1% of the PSMs are not true positives. It is not an individual parameter and several PSMs can have a much higher probability of being false than 1%. I added a second paramter, which is the PEP based Q-value. It's the last column. In that column, you want all PSMs with a q-value less that 0.01. That is a set-wise parameter and comparable to the old Q-value.